torchgen/gen_functionalization_type.py

```diff
@@ -736,6 +738,9 @@ def emit_registration_helper(f: NativeFunction) -> str:
         if str(f.func.name) == "resize_":
             # See Note [resize_ in Functionalization]
             return []
+        if str(f.func.name.name) == "set_":
+            # set_ is a crazy op that swaps out storage
+            return []
```

Comment:
What happens if you remove this?

I think the problem is that we are no longer codegen'ing functionalziation kernels for any of the set_() ops, which we still want to do here

Reply:
@bdhirsh I added this in an attempt to get around this build failure:

```
cd /home/slarsen/local/pytorch && /home/slarsen/local/miniconda3/envs/pytorch-3.10/bin/python -m torchgen.gen --source-path /home/slarsen/local/pytorch/cmake/../aten/src/ATen --install_dir /home/slarsen/local/pytorch/build/aten/src/ATen --per-operator-headers --generate sources --output-dependencies /home/slarsen/local/pytorch/build/aten/src/ATen/generated_sources.cmake
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/slarsen/local/pytorch/torchgen/gen.py", line 2880, in <module>
    main()
  File "/home/slarsen/local/pytorch/torchgen/gen.py", line 2820, in main
    gen_source_files(
  File "/home/slarsen/local/pytorch/torchgen/gen.py", line 2518, in gen_source_files
    cpu_fm.write_sharded(
  File "/home/slarsen/local/pytorch/torchgen/utils.py", line 237, in write_sharded
    env = env_callable(item)
          ^^^^^^^^^^^^^^^^^^
  File "/home/slarsen/local/pytorch/torchgen/gen.py", line 2490, in functionalization_env_callable
    "func_registrations": gen_functionalization_registration(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/slarsen/local/pytorch/torchgen/gen_functionalization_type.py", line 744, in gen_functionalization_registration
    assert not f.is_view_op
AssertionError
```

Reply:
(hopefully) fixed offline